### PR TITLE
Add clippy to CI

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Add clippy task to CI.

For now only these lints will cause a CI failure, the rest will simply be warnings (so this will only prevent merge for obviously wrong code):

```
rustflags = [
  "-Dunsafe_code",
  "-Dclippy::correctness",
  "-Dclippy::complexity",
  "-Dclippy::suspicious",
]
```

In the future we should add more lints, and deny warnings.

### CODE REVIEW

File: runtime/test/src/lib.rs [hunk 1]
```diff
@@ -424,7 +424,7 @@ impl parachain_info::Config for Runtime {}
 impl cumulus_pallet_aura_ext::Config for Runtime {}
 
 parameter_types! {
-    pub const Period: u32 = prod_or_fast!(6 * HOURS, 1 * MINUTES);
+    pub const Period: u32 = prod_or_fast!(6 * HOURS, MINUTES);
```
Comment (tmpolaczyk) [inline runtime/test/src/lib.rs:427]: Revert this one

File: pallets/authority-assignment/src/mock.rs [hunk 2]
```diff
@@ -106,20 +106,12 @@ pub mod mock_data {
 
 #[derive(Clone, Encode, Decode, PartialEq, sp_core::RuntimeDebug, scale_info::TypeInfo)]
 #[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Default)]
```
Comment (tmpolaczyk) [inline pallets/authority-assignment/src/mock.rs:109]: fmt

File: node/src/container_chain_spawner.rs [hunk 3]
```diff
@@ -97,7 +97,7 @@ impl ContainerChainSpawner {
             spawned_para_ids,
             collate_on_tanssi: _,
         } = self.clone();
-        let mut container_chain_cli: ContainerChainCli = container_chain_cli.clone();
+        let mut container_chain_cli: ContainerChainCli = container_chain_cli;
```
Comment (tmpolaczyk) [inline node/src/container_chain_spawner.rs:100]: Remove type annotation

File: container-chains/templates/simple/node/src/service.rs [hunk 4]
```diff
@@ -246,7 +246,7 @@ async fn start_node_impl(
         .map_err(|e| sc_service::Error::Application(Box::new(e)))?;
 
     let announce_block = {
-        let sync_service = sync_service.clone();
+        let sync_service = sync_service;
```
Comment (tmpolaczyk) [inline container-chains/templates/simple/node/src/service.rs:249]: Maybe not needed
```suggestion
```

**Review by girazoki — APPROVED** [review]
Fully support, specially given all the things that it brought up..

**Comment by girazoki** [comment]
Feel free to merge now @tmpolaczyk

**Pull request merged**